rsrcdump/packutils.py

```diff
@@ -1,48 +1,50 @@
+from typing import Tuple, Union
+
+from io import BytesIO
 import struct
 
 class Unpacker:
-    def __init__(self, data, offset=0):
+    def __init__(self, data: bytes, offset: int=0) -> None:
         self.data = data
-        self.offset = 0
-    
-    def unpack(self, fmt):
+        self.offset = offset
+
+    def unpack(self, fmt: str) -> Tuple:
         record_length = struct.calcsize(fmt)
         fields = struct.unpack_from(fmt, self.data, self.offset)
         self.offset += record_length
         return fields
 
-    def seek(self, offset):
+    def seek(self, offset: int) -> None:
         self.offset = offset
 
-    def read(self, size):
+    def read(self, size: int) -> bytes:
         data_slice = self.data[self.offset : self.offset + size]
         assert len(data_slice) == size
         self.offset += size
         return data_slice
 
-    def unpack_pstr(self, decode=True):
+    def unpack_pstr(self, decode: bool=True) -> Union[str, bytes]:
         length, = self.unpack(">B")
         binary_pstr = self.read(length)
         if decode:
             return binary_pstr.decode("macroman")
-        else:
-            return binary_pstr
+        return binary_pstr
 
-    def eof(self):
+    def eof(self) -> bool:
         return self.offset >= len(self.data)
 
-    def remaining(self):
+    def remaining(self) -> int:
         return len(self.data) - self.offset
 
 class WritePlaceholder:
-    def __init__(self, stream, fmt):
+    def __init__(self, stream: BytesIO, fmt: str) -> None:
         self.stream = stream
         self.fmt = fmt
         self.position = self.stream.tell()
         self.stream.write(b'\xCA' * struct.calcsize(fmt))
         self.committed = False
 
-    def commit(self, value):
+    def commit(self, value: Union[int, bytes, str]) -> None:
```

Comment:
Let's make `value` an `Any` to match the signature of `struct.pack`.
```suggestion
    def commit(self, value: Any) -> None:
```